elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/dynamic/DynamicAccessorValidator.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.converters.avro.dynamic
+
+import com.spotify.elitzur.MetricsReporter
+import com.spotify.elitzur.converters.avro.dynamic.dsl.AvroObjMapper
+import com.spotify.elitzur.validators.{DynamicRecordValidator, Unvalidated, Validator}
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+
+import scala.util.{Failure, Success, Try}
+
+class DynamicAccessorValidator(
+  recordAccessorWithValidator: Array[(String, DynamicValidationCompanion)],
+  schema: Schema
+)(implicit metricsReporter: MetricsReporter) {
+
+  final val className: String = this.getClass.getName
+
+  // From the user provided input, create a parser that can extract a value from a record and apply
+  // to it the the parsing rule defined in the companion object.
+  private[elitzur] val fieldParsers: Array[DynamicFieldParser] = {
```

Comment:
Bit a nitpick. I think we can extract all the errors at once and build a better error message if more than one thing is invalid. I don't really see this as necessary so if it's too much effort I'm ok with the current way this is implemented

Reply:
I suppose if there's a lot of nested fields this might just be a tonne of noise if something is wrong at the root

Reply:
It shouldn't be complex to implement, will update the pr with the suggested change.

Reply:
change made to collect all the errors, rather than failing on first